tests/test_platform_config.py

```diff
@@ -0,0 +1,320 @@
+        assert config.device_string == "xpu"
+
+    def test_get_platform_invalid_raises_valueerror(self):
+        """get_platform() with invalid name should raise ValueError."""
+        with pytest.raises(ValueError) as exc_info:
+            get_platform("invalid_platform")
+        assert "Unknown platform" in str(exc_info.value)
+        assert "invalid_platform" in str(exc_info.value)
+
+    def test_get_platform_error_shows_available(self):
+        """ValueError message should list available platforms."""
+        with pytest.raises(ValueError) as exc_info:
+            get_platform("nonexistent")
+        error_msg = str(exc_info.value)
+        assert "cuda" in error_msg
+        assert "xpu" in error_msg
+
+    def test_get_platform_case_sensitive(self):
+        """Platform names should be case-sensitive."""
+        with pytest.raises(ValueError):
+            get_platform("CUDA")
+        with pytest.raises(ValueError):
+            get_platform("Xpu")
+
+
+class TestGetPlatformChoices:
+    """Tests for the get_platform_choices() function."""
+
+    def test_get_platform_choices_returns_list(self):
+        """get_platform_choices() should return a list."""
+        choices = get_platform_choices()
+        assert isinstance(choices, list)
+
+    def test_get_platform_choices_contains_cuda_and_xpu(self):
+        """Choices should include both cuda and xpu."""
+        choices = get_platform_choices()
+        assert "cuda" in choices
+        assert "xpu" in choices
+
+    def test_get_platform_choices_is_sorted(self):
+        """Choices should be sorted alphabetically."""
+        choices = get_platform_choices()
+        assert choices == sorted(choices)
+
+    def test_get_platform_choices_matches_platforms_dict(self):
+        """Choices should match PLATFORMS registry keys."""
+        choices = get_platform_choices()
+        assert set(choices) == set(PLATFORMS.keys())
+
+
+class TestCudaPlatformConfig:
+    """Tests specific to CUDA platform configuration."""
+
+    def test_cuda_device_string(self):
+        """CUDA device_string should be 'cuda'."""
+        config = get_platform("cuda")
+        assert config.device_string == "cuda"
+
+    def test_cuda_has_empty_guidance(self):
+        """CUDA should have empty guidance (no special instructions needed)."""
+        config = get_platform("cuda")
+        assert config.guidance_block == ""
+        assert config.kernel_guidance == ""
+
+    def test_cuda_has_no_hacks_to_strip(self):
+        """CUDA should have no CUDA hacks to strip."""
+        config = get_platform("cuda")
+        assert config.cuda_hacks_to_strip == ()
+        assert len(config.cuda_hacks_to_strip) == 0
+
+
+class TestXpuPlatformConfig:
+    """Tests specific to XPU platform configuration."""
+
+    def test_xpu_device_string(self):
+        """XPU device_string should be 'xpu'."""
+        config = get_platform("xpu")
+        assert config.device_string == "xpu"
```

Comment:
Some of the tests here are duplicates of those in `TestTargetPlatformPromptBlock`

Let's combine and keep them here